core/src/banking_stage.rs

```diff
@@ -543,154 +527,102 @@ impl BankingStage {
-                        let (process_transactions_summary, process_packets_transactions_time) =
-                            Measure::this(
-                                |_| {
-                                    Self::process_packets_transactions(
-                                        &working_bank,
-                                        &bank_creation_time,
-                                        recorder,
-                                        packet_batch,
-                                        original_unprocessed_indexes.to_owned(),
-                                        transaction_status_sender.clone(),
-                                        gossip_vote_sender,
-                                        banking_stage_stats,
-                                        qos_service,
-                                        slot_metrics_tracker,
-                                    )
-                                },
-                                (),
-                                "process_packets_transactions",
-                            );
-                        slot_metrics_tracker.increment_process_packets_transactions_us(
-                            process_packets_transactions_time.as_us(),
+                    banking_stage_stats
+                        .end_of_slot_filtered_invalid_count
+                        .fetch_add(end_of_slot_filtered_invalid_count, Ordering::Relaxed);
+
+                    deserialized_packet_batch.update_buffered_packets_with_new_unprocessed(
+                        &original_unprocessed_indexes,
+                        &new_unprocessed_indexes,
+                    )
+                } else {
+                    true
+                };
+                end_of_slot_filtering_time.stop();
+                slot_metrics_tracker
+                    .increment_end_of_slot_filtering_us(end_of_slot_filtering_time.as_us());
+                should_retain
+            } else {
+                let (bank_start, poh_recorder_lock_time) = Measure::this(
+                    |_| poh_recorder.lock().unwrap().bank_start(),
+                    (),
+                    "poh_recorder_lock",
+                );
+                slot_metrics_tracker.increment_consume_buffered_packets_poh_recorder_lock_us(
+                    poh_recorder_lock_time.as_us(),
+                );
+
+                if let Some(BankStart {
+                    working_bank,
+                    bank_creation_time,
+                }) = bank_start
+                {
+                    let (process_transactions_summary, process_packets_transactions_time) =
+                        Measure::this(
+                            |_| {
+                                Self::process_packets_transactions(
+                                    &working_bank,
+                                    &bank_creation_time,
+                                    recorder,
+                                    packet_batch,
```

Comment:
can we pass the `deserialize_packet_batch.unprocessed_packets` here instead so we don't have to incur the cost of deserializing these packets again?

Reply:
Similar to above, `process_packets_transactions` is going to be replace by `process_transactions(&transactions...)`, to take advantage of packets were deserialized upon receiving.